packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -131,20 +127,29 @@ - (void)cleanupWithCompletion:(void (^)(void))completion {
   // Cleanup credentials.
   [_credentials removeAllObjects];
 
+  for (FlutterEventChannel *channel in self->_eventChannels) {
+    [channel setStreamHandler:nil];
+  }
+  [self->_eventChannels removeAllObjects];
+  for (NSObject<FlutterStreamHandler> *handler in self->_streamHandlers) {
+    [handler onCancelWithArguments:nil];
+  }
+  [self->_streamHandlers removeAllObjects];
+
   if (completion != nil) completion();
 }
 
 - (void)detachFromEngineForRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [self cleanupWithCompletion:nil];
-  self.channel = nil;
 }
 
 - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)flutterResult {
   FLTFirebaseMethodCallErrorBlock errorBlock =
       ^(NSString *_Nullable code, NSString *_Nullable message, NSDictionary *_Nullable details,
         NSError *_Nullable error) {
         if (code == nil) {
-          NSDictionary *errorDetails = [self getNSDictionaryFromNSError:error];
+          NSDictionary *errorDetails = [FLTFirebaseAuthPlugin getNSDictionaryFromNSError:error];
+          [self storeAuthCredentialIfPresent:error];
```

Comment:
This was extracted from `getNSDictionaryFromNSError` in order to make `getNSDictionaryFromNSError` static. 

I think the "getNSDictionaryFrom*" methods should live in a static utility class to enforce SRP.